### PR TITLE
Fix wrong association bug and add edit category functionality

### LINKED ISSUE #99: Expense not associating with Category on startup
Expenses are not associating properly with the Category objects upon startup.

### PR DESCRIPTION
**Pull request by gitsac**
Closes #99 , #78

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditCategory.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ */
+public class EditCategory extends Command {
+    public static final String COMMAND_WORD = "ecat";
+
+    public static final String MESSAGE_NO_EXISTING_CATEGORY = "There is no such category!";
+
+    public static final String MESSAGE_DEFAULT_CATEGORY = "The MISC category can not be edited!";
+    private final Index targetIndex;
+
+    private final String newCategoryName;
```
Comment (shirsho-12) [inline src/main/java/seedu/address/logic/commands/EditCategory.java:26]: follows command structure

File: src/main/java/seedu/address/commons/core/Messages.java [hunk 2]
```diff
@@ -10,6 +10,10 @@ public class Messages {
 
     public static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format! \n%1$s";
     public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The person index provided is invalid";
+    public static final String MESSAGE_INVALID_CATEGORY_DISPLAYED_INDEX = "The category index provided is invalid!";
+    public static final String MESSAGE_INVALID_EDIT_FOR_CATEGORIES = "Please specify an edit to at least "
+            + "the category name or the category's summary.";
+    public static final String MESSAGE_SUCCESSFULLY_EDITED_CATEGORY = "Edited category: %1$s";
```
Comment (shirsho-12) [inline src/main/java/seedu/address/commons/core/Messages.java:16]: nice, follows structure

**Review by shirsho-12 — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/EditCategory.java (hunk 1 above)
Comment (randallnhr) [reply src/main/java/seedu/address/logic/commands/EditCategory.java:26]: Should we edit the command for categories to acat - add category, dcat to delete category to follow the formatting of ecat and lcat?

**Review by randallnhr — COMMENTED** [review]
LGTM!

**Pull request merged**